CONTRIBUTING.md

```diff
@@ -50,109 +35,191 @@ In order for a PR to be merged, it must pass the following requirements:
 - The PR must be approved by at least one maintainer
   - The PR must be approved by 2+ maintainers if the PR is a new feature or > 100 LOC changed
 
-## Branching
+## Adding a new repository
 
-This section outlines the branching strategy of this repo.
+### 1. Add the repository to packages
 
-### Main
+Run `git submodule add https://github.com/Uniswap/<repository-name> src/pkgs/<repository-name>`
 
-The main branch is supposed to reflect the deployed state on all networks. Any pull requests into this branch MUST come from the staging branch.
+### 2. Add compilation settings to foundry.toml
 
-### Staging
+In the `foundry.toml` file, first create a profile with the compiler settings.
 
-The staging branch reflects new code complete deployments or upgrades containing fixes and/or features. Any pull requests into this branch MUST come from the dev branch. The staging branch is used for security audits and deployments. Once the deployment is complete and verified as well as deployment log files are generated, the branch can be merged into main. For more information on the deployment and log file generation check [here](#deployment).
+```
+additional_compiler_profiles = [
+  ...
+  { name = "<repository-name>", optimizer_runs = <optimizer-runs>, via_ir = <via_ir>, ... },
+]
+```
 
-### Dev
+Next, add the compilation restrictions to use the new profile.
```

Comment:
a blurb here about how it uses the restrictions to find a matching profile would help connect the dots